### PR TITLE
Fixes #500

### LINKED ISSUE #500: Create Internationalization Integration Test
Related to #427

Because there was confusion as to how to setup internationalization, this will be valuable in making it clear.

### PR DESCRIPTION
**Pull request by jarias**
Create Internationalization Integration Test

#500

### CODE REVIEW

File: examples/servlet/pom.xml [hunk 1]
```diff
@@ -31,6 +32,12 @@
         <logback.version>1.1.2</logback.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <servlet.version>3.0.1</servlet.version>
+        <groovy.version>1.8.8</groovy.version>
+        <testng.version>6.8</testng.version>
+        <restassured.version>2.8.0</restassured.version>
```
Comment (mrioan) [inline examples/servlet/pom.xml]: Can you please move newly added dependencies to the root pom? They should be referenced here but not added.

File: examples/servlet/pom.xml [hunk 2]
```diff
@@ -56,11 +63,83 @@
...
+                            <goal>testCompile</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <invokeDynamic>true</invokeDynamic>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
Comment (mrioan) [inline examples/servlet/pom.xml]: We already have surefire and failsafe in the root pom, didn't it work here? It should and therefore these plugins should not be required here. Take a look at `stormpath-spring-security-webmvc`. That project hast ITs running but the plugin does not show up in its `pom.xml`

File: examples/servlet/pom.xml [hunk 3]
```diff
@@ -92,4 +194,17 @@
         </plugins>
     </build>
 
+    <profiles>
```
Comment (mrioan) [inline examples/servlet/pom.xml]: We do not follow this procedure in the other modules. We basically have defined a project wide `skiptITs` so ITs never run unless explicitly adding `skipITs=true` when executing maven. This may be a good approach but we need to define this project-wise. Therefore, for now please remove the profile. Thanks!

File: examples/servlet/src/test/groovy/com/stormpath/sdk/examples/servlet/i18n/I18nIT.groovy [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import org.testng.annotations.Test
+
+import static com.jayway.restassured.RestAssured.given
+import static org.hamcrest.Matchers.equalTo
+
+/**
+ *
+ */
+class I18nIT {
+    @Test
```
Comment (mrioan) [inline examples/servlet/src/test/groovy/com/stormpath/sdk/examples/servlet/i18n/I18nIT.groovy]: Blank line here please

File: examples/servlet/src/test/groovy/com/stormpath/sdk/examples/servlet/i18n/I18nIT.groovy [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+import com.jayway.restassured.RestAssured
+import com.jayway.restassured.config.DecoderConfig
+import org.testng.annotations.Test
+
+import static com.jayway.restassured.RestAssured.given
+import static org.hamcrest.Matchers.equalTo
+
+/**
+ *
```
Comment (mrioan) [inline examples/servlet/src/test/groovy/com/stormpath/sdk/examples/servlet/i18n/I18nIT.groovy]: `@since 1.0.XX` here please

File: examples/servlet/src/test/groovy/com/stormpath/sdk/examples/servlet/i18n/I18nIT.groovy [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                .header("Accept-Language", "es")
+                .when()
+                .get("http://localhost:8080/register")
+                .then()
+                .body(
+                "html.head.title", equalTo("Crear Cuenta"),
+                "**.findAll { it.@type == 'submit' }", equalTo("Crear Cuenta"),
+                "**.findAll { it.@class == 'to-login' }", equalTo("Regresar a Iniciar Sesión")
+        )
+    }
```
Comment (mrioan) [inline examples/servlet/src/test/groovy/com/stormpath/sdk/examples/servlet/i18n/I18nIT.groovy]: Please add another test asserting that by default (no Accept-language) the language texts are displayed in English

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/i18n/DefaultMessageSource.java [hunk 7]
```diff
@@ -36,12 +37,15 @@ public String getMessage(String key, Locale locale) {
...
         ResourceBundle bundle = getBundle(locale);
-        String msg = bundle.getString(key);
 
         try {
+            String msg = new String(bundle.getString(key).getBytes("ISO-8859-1"), "UTF-8");
             return MessageFormat.format(msg, args);
         } catch (MissingResourceException e) {
             return '!' + key + '!';
+        } catch (UnsupportedEncodingException e) {
+            //Should not happen
```
Comment (mrioan) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/i18n/DefaultMessageSource.java]: can you elaborate that comment please?: "Should not happen because... "

**Comment by mrioan** [comment]
@jarias First round of review ready, back to you. Thanks!

File: examples/servlet/pom.xml (hunk 1 above)
Comment (jarias) [inline examples/servlet/pom.xml]: There is no root pom for the examples, I could move the test to the servlet module instead but then It would have a dependency on the servlet example war to be able to the start the app.

File: examples/servlet/pom.xml (hunk 2 above)
Comment (jarias) [inline examples/servlet/pom.xml]: Same as above the examples don't have a root pom.

File: examples/servlet/src/test/groovy/com/stormpath/sdk/examples/servlet/i18n/I18nIT.groovy (hunk 6 above)
Comment (jarias) [inline examples/servlet/src/test/groovy/com/stormpath/sdk/examples/servlet/i18n/I18nIT.groovy]: Sure thing

File: examples/servlet/pom.xml (hunk 1 above)
Comment (lhazlewood) [inline examples/servlet/pom.xml]: I think @mrioan is saying that all dependencies - no matter what for - are declared in the project root pom.xml and referenced by submodules.  This better ensures that two different submodules do not accidentally pull in different versions of the same library because the module author can always check the root project pom to see what is globally referenceable

Comment (lhazlewood) [inline examples/servlet/pom.xml]: In other words, the examples/servlet/pom.xml should even have a `<properties>` section that defines library versions since those libs/versions should be declared in the root pom.

If a submodule needs a different version of a library for a very specific reason - perhaps that example is not compatible with the servlet API used in the rest of the project (for example), only then would it make sense to declare that particular version in the local pom.xml file IMO.

File: examples/servlet/pom.xml (hunk 2 above)
Comment (lhazlewood) [inline examples/servlet/pom.xml]: They do have a root pom - the examples pom extends from the project root pom.

File: examples/servlet/src/test/groovy/com/stormpath/sdk/examples/servlet/i18n/I18nIT.groovy [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+                    .body(
+                        "html.head.title", equalTo("Iniciar Sesión"),
+                        "**.findAll { it.@class == 'header' }.span.a", equalTo("Crear Cuenta"),
+                        "**.findAll { it.@type == 'submit' }", equalTo("Iniciar Sesión"),
+                        "**.findAll { it.@class == 'to-login' }", equalTo("Se te olvidó tu contraseña?")
+                    )
+    }
+
+    @Test
+    void testRegisterPage() {
```
Comment (lhazlewood) [inline examples/servlet/src/test/groovy/com/stormpath/sdk/examples/servlet/i18n/I18nIT.groovy]: Very nice test!  I like the `findAll` approach of verifying body contents :)

File: examples/servlet/pom.xml (hunk 2 above)
Comment (jarias) [inline examples/servlet/pom.xml]: Yeah but the the servlet example doesn't have a parent.

Comment (lhazlewood) [inline examples/servlet/pom.xml]: Ahhhh - interesting.  @mrioan and @dogeared is this because we wanted to run them standalone for copy-paste reasons (so other people didn't have to depend on our SDK pom?)

Comment (mrioan) [inline examples/servlet/pom.xml]: @lhazlewood That was the original idea. It seems that if we want to run ITs directly in the examples then we will need a root pom for the examples. This way we will still be able to depend on our jar files (rather than on the source code) while having all plugins/dependencies available in the root pom of the examples.

The question is, can the examples' root pom have the SDK as parent (as it currently is)?

In summary: example1, example2 and example3 should examples/pom.xml as their parent. But, is it really possible? Right now examples/pom.xml has the root SDK as parent. So, by transitive nature, example1, example2, example3, will then have the root SDK pom as parent as well?

Comment (lhazlewood) [inline examples/servlet/pom.xml]: What was the outcome of the standup today?  Will we have a shared parent or not for the examples?

**Comment by lhazlewood** [comment]
@jarias build is failing - can you please investigate?

**Comment by jarias** [comment]
@lhazlewood yeah I'm still doing some local changes, the outcome was to depend on the examples pom instead of the root pom and publish that examples pom so people can run the examples standalone, at the same time sharing configuration and dependencies between examples

**Comment by dogeared** [comment]
LGTM. After CI passes, please merge.
